### ISSUE TITLE
Regression: `could not convert tokens` in some macros

### ISSUE THREAD

**Issue by rscarson** [rust-lang/rust-analyzer#17701]
**rust-analyzer version**: v0.3.2045

**rustc version**: rustc 1.79.0 (129f3b996 2024-06-10)

**editor or extension**: VSCode

**code snippet to reproduce**: 
```rust
extension!(
    init_console,
    deps = [rustyscript],
    esm_entry_point = "ext:init_console/init_console.js",
    esm = [ dir "src/ext/console", "init_console.js" ],
);
```
![image](https://github.com/user-attachments/assets/e22ef73c-fff6-4226-974c-8eb2417d7df0)

Upstream issue: https://github.com/denoland/deno_core/issues/844

The macro in question: https://github.com/denoland/deno_core/blob/85b40e5d47e568563957b031b5c3d4673226cc20/core/extensions.rs#L386

**Comment by maxdeviant** [comment]
I started seeing this recently (in the last day or so) with the [`wasmtime::component::bindgen!`](https://docs.rs/wasmtime/21.0.1/wasmtime/component/macro.bindgen.html) macro.

I'm seeing it in all usages of the macro in the Zed repo:

https://github.com/zed-industries/zed/blob/fe23504eba5db02f8f5796e0c97cc328794b024f/crates/extension/src/wasm_host/wit/since_v0_0_7.rs#L27-L35

**Comment by Veykril** [comment]
Ugh our `()` wrapping expressions bites us here, the decl macro chain ends up with us wrapping the file path in `()` which breaks the builtin `include_str` parser as it expects a literal

**Comment by Veykril** [comment]
#17706 should workaround this issue for now

**Comment by itowlson** [comment]
I'm still seeing this in 0.3.2053, even though the changelog says it includes the #17706 fix.  I did a full uninstall-reinstall of the VS Code extension and created a fresh project (in case there was some cached state), but still the same error.

(I also tried 0.3.2054, but that only changed the error to "Expected string literal.")

An example that exhibits the error for me is https://github.com/itowlson/ra-macro-errors.  Thanks!
